cmds/fmap/fmap.go

```diff
@@ -29,10 +35,17 @@ import (
 )
 
 var (
-	summary = flag.Bool("s", false, "print human readable summary")
-	read    = flag.Int("r", -1, "read an area from the flash")
+	checksum = flag.String("c", "", "print checksum using the given `hash function` (md5|sha1|sha256)")
+	read     = flag.Int("r", -1, "read an area from the flash")
+	summary  = flag.Bool("s", false, "print human readable summary")
 )
 
+var hashFuncs = map[string]hash.Hash{
+	"md5":    md5.New(),
```

Comment:
Do you actually want to call all these functions? Or just put md5.New here and call it when you choose the hash?